### PR TITLE
ROX-20605, ROX-20606: Make requester and approver searchable

### PR DESCRIPTION
**Pull request by md2119**
## Description

This PR makes changes to the data model to make the requester and approver searchable and adds a migration for the new data model. 

In v2, we have a requirement to be able to search by requester and approver names. However, it could not be done with the existing data model. This is because the `requester` and `approvers` fields shared the same type `storage.SlimUser` and we do not support different search labels on the same message or message hierarchy.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Unit + Manual

```
$ curl -sSk -u "admin:e9s99uQzFToVYSaaWowYetAsM" https://104.154.138.198/v1/cve/requests | jq
{
  "requestInfos": [
    {
      "id": "d0bc53f8-e0bd-4532-a699-45b515a6fb04",
      "name": "AA-231101-1",
      "targetState": "DEFERRED",
      "status": "APPROVED",
      "expired": false,
      "requestor": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [
        {
          "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
          "name": "admin"
        }
      ],
      "createdAt": "2023-11-01T15:42:06.526413935Z",
      "lastUpdated": "2023-11-01T23:00:12.600081848Z",
      "comments": [
        {
          "id": "90db1e85-064f-4b8d-a674-ab5a5d122ae9",
          "message": "defer",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T15:42:06.526431134Z"
        },
        {
          "id": "f09ce9d1-8bd1-410f-b7b2-a9c8873374b0",
          "message": "approved\n",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T23:00:12.600091431Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "gke.gcr.io",
          "remote": "calico/node",
          "tag": "v3.23.5-gke.10"
        }
      },
      "deferralReq": {
        "expiry": {
          "expiresWhenFixed": true,
          "expiryType": "TIME"
        }
      },
      "cves": {
        "cves": [
          "RHSA-2023:5455"
        ]
      }
    },
    {
      "id": "874550d1-d208-46c5-8780-c02dfa425269",
      "name": "AA-231101-2",
      "targetState": "DEFERRED",
      "status": "PENDING",
      "expired": false,
      "requestor": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [],
      "createdAt": "2023-11-01T15:42:14.572726897Z",
      "lastUpdated": "2023-11-01T15:42:14.572726897Z",
      "comments": [
        {
          "id": "6d8f8b90-bbd9-4af0-ac03-e0b10097e86f",
          "message": "defer",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T15:42:14.572740685Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "gke.gcr.io",
          "remote": "calico/node",
          "tag": ".*"
        }
      },
      "deferralReq": {
        "expiry": {
          "expiresWhenFixed": true,
          "expiryType": "TIME"
        }
      },
      "cves": {
        "cves": [
          "CVE-2021-35937"
        ]
      }
    },
    {
      "id": "ca93899f-ca34-4857-b127-90d1aaacece4",
      "name": "AA-231101-3",
      "targetState": "FALSE_POSITIVE",
      "status": "APPROVED",
      "expired": false,
      "requestor": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [
        {
          "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
          "name": "admin"
        }
      ],
      "createdAt": "2023-11-01T22:59:49.263754985Z",
      "lastUpdated": "2023-11-01T23:00:18.940219419Z",
      "comments": [
        {
          "id": "69455fa3-ee75-47a7-b7b7-9a3723892fcf",
          "message": "false positive",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T22:59:49.263776648Z"
        },
        {
          "id": "0cc9585d-9ede-4c2a-ab1e-196f0400690d",
          "message": "approved",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T23:00:18.940228612Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "quay.io",
          "remote": "stackrox-io/central-db",
          "tag": ".*"
        }
      },
      "fpRequest": {},
      "cves": {
        "cves": [
          "CVE-2007-4559"
        ]
      }
    },
    {
      "id": "e9720e3e-11dd-45e7-bd50-e109249a936e",
      "name": "AA-231101-4",
      "targetState": "DEFERRED",
      "status": "PENDING",
      "expired": false,
      "requestor": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [],
      "createdAt": "2023-11-01T23:00:00.854599976Z",
      "lastUpdated": "2023-11-01T23:00:00.854599976Z",
      "comments": [
        {
          "id": "5fa5a0b5-e9b8-4e2d-bd8f-8b484b299e1a",
          "message": "defer",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T23:00:00.854609382Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "quay.io",
          "remote": "stackrox-io/central-db",
          "tag": ".*"
        }
      },
      "deferralReq": {
        "expiry": {
          "expiresWhenFixed": true,
          "expiryType": "TIME"
        }
      },
      "cves": {
        "cves": [
          "CVE-2016-4607"
        ]
      }
    }
  ]
}



$ kubectl logs deploy/central -n stackrox -f
No certificates found in /usr/local/share/ca-certificates
No certificates found in /etc/pki/injected-ca-trust
move-to-current: 2023/11/01 16:54:55.796302 logger.go:61: Info: Move databases to current
Migrator: 2023/11/01 16:54:59.031174 log.go:18: Info: Run migrator.run() with version: 4.3.x-25-gca713734e4, DB sequence: 195
Migrator: 2023/11/01 16:54:59.033310 log.go:18: Info: conf.Maintenance.ForceRollbackVersion: none
Migrator: 2023/11/01 16:54:59.037049 log.go:18: Info: checking if the database "central_active" exists
clone/rocksdb: 2023/11/01 16:54:59.168276 db_clone_manager_impl.go:60: Info: Found clone current -> .db-init
clone/rocksdb: 2023/11/01 16:54:59.168372 db_clone_manager_impl.go:112: Info: Database clones:
clone/rocksdb: 2023/11/01 16:54:59.168498 db_clone_manager_impl.go:114: Info: current -> &{/var/lib/stackrox/current 0 0 0 0001-01-01 00:00:00 +0000 UTC}
Migrator: 2023/11/01 16:54:59.168605 log.go:18: Info: connect to gorm: "host=central-db.stackrox.svc port=5432 user=postgres sslmode=verify-full sslrootcert=/run/secrets/stackrox.io/certs/ca.pem statement_timeout=1.2e+06 pool_min_conns=10 pool_max_conns=90 client_encoding=UTF8"
Migrator: 2023/11/01 16:54:59.342119 log.go:18: Info: Migration version from DB = seq_num:195 version:"4.3.x-25-gca713734e4" last_persisted:<seconds:1698767659 nanos:82742000 > min_seq_num:186 .
clone/postgres: 2023/11/01 16:54:59.343877 db_clone_manager_impl.go:96: Info: db is of version &{ 4.3.x-25-gca713734e4 195 186 2023-10-31 15:54:19.082742 +0000 UTC}
clone/postgres: 2023/11/01 16:54:59.425342 db_clone_manager_impl.go:169: Info: safeRemove -> central_temp
NOTICE:  database "central_temp" does not exist, skipping
clone/postgres: 2023/11/01 16:54:59.933223 db_clone_manager_impl.go:160: Info: Postgres Database clones:
clone/postgres: 2023/11/01 16:54:59.933353 db_clone_manager_impl.go:162: Info: central_active -> &{ 4.3.x-25-gca713734e4 195 186 2023-10-31 15:54:19.082742 +0000 UTC}
clone/postgres: 2023/11/01 16:54:59.933377 db_clone_manager_impl.go:225: Info: GetCloneToMigrate
clone/postgres: 2023/11/01 16:54:59.933461 db_clone_manager_impl.go:300: Info: Fell through all checks to return current.
Migrator: 2023/11/01 16:54:59.933481 log.go:18: Info: Clone to Migrate "", "central_active"
Migrator: 2023/11/01 16:54:59.933502 log.go:13: Info: starting DB compaction
Migrator: 2023/11/01 16:54:59.933556 log.go:13: Info: no existing DB exists. Stopping DB compaction
Migrator: 2023/11/01 16:54:59.933571 log.go:18: Info: Load database = "central_active"
Migrator: 2023/11/01 16:55:00.024431 log.go:13: Info: Successfully connected to central database.
Migrator: 2023/11/01 16:55:00.096234 log.go:18: Info: Migration version from DB = seq_num:195 version:"4.3.x-25-gca713734e4" last_persisted:<seconds:1698767659 nanos:82742000 > min_seq_num:186 .
Migrator: 2023/11/01 16:55:00.096481 log.go:18: Info: version for "central_active" is &{ 4.3.x-25-gca713734e4 195 186 2023-10-31 15:54:19.082742 +0000 UTC}
Migrator: 2023/11/01 16:55:00.096556 log.go:18: Info: In runner.Run
Migrator: 2023/11/01 16:55:00.142124 log.go:18: Info: Migration version from DB = seq_num:195 version:"4.3.x-25-gca713734e4" last_persisted:<seconds:1698767659 nanos:82742000 > min_seq_num:186 .
Migrator: 2023/11/01 16:55:00.142229 log.go:18: Info: DB is up to date at version 195. Nothing to do here.

$ kubectl set image deploy/central central=quay.io/stackrox-io/main:4.3.x-50-gc5ba2de4e3 -n stackrox
deployment.apps/central image updated

$ kubectl logs deploy/central -n stackrox -f
No certificates found in /usr/local/share/ca-certificates
No certificates found in /etc/pki/injected-ca-trust
move-to-current: 2023/11/01 23:04:49.115229 logger.go:61: Info: Move databases to current
Migrator: 2023/11/01 23:04:49.500633 log.go:18: Info: Run migrator.run() with version: 4.3.x-50-gc5ba2de4e3, DB sequence: 196
Migrator: 2023/11/01 23:04:49.501427 log.go:18: Info: conf.Maintenance.ForceRollbackVersion: none
Migrator: 2023/11/01 23:04:49.503655 log.go:18: Info: checking if the database "central_active" exists
clone/rocksdb: 2023/11/01 23:04:49.555675 db_clone_manager_impl.go:60: Info: Found clone current -> .db-init
clone/rocksdb: 2023/11/01 23:04:49.556447 db_clone_manager_impl.go:112: Info: Database clones:
clone/rocksdb: 2023/11/01 23:04:49.556760 db_clone_manager_impl.go:114: Info: current -> &{/var/lib/stackrox/current 0 0 0 0001-01-01 00:00:00 +0000 UTC}
Migrator: 2023/11/01 23:04:49.556978 log.go:18: Info: connect to gorm: "host=central-db.stackrox.svc port=5432 user=postgres sslmode=verify-full sslrootcert=/run/secrets/stackrox.io/certs/ca.pem statement_timeout=1.2e+06 pool_min_conns=10 pool_max_conns=90 client_encoding=UTF8"
Migrator: 2023/11/01 23:04:49.655530 log.go:18: Info: Migration version from DB = seq_num:195 version:"4.3.x-25-gca713734e4" last_persisted:<seconds:1698857710 nanos:186140000 > min_seq_num:186 .
clone/postgres: 2023/11/01 23:04:49.655954 db_clone_manager_impl.go:96: Info: db is of version &{ 4.3.x-25-gca713734e4 195 186 2023-11-01 16:55:10.18614 +0000 UTC}
clone/postgres: 2023/11/01 23:04:49.731970 db_clone_manager_impl.go:169: Info: safeRemove -> central_temp
NOTICE:  database "central_temp" does not exist, skipping
clone/postgres: 2023/11/01 23:04:49.932784 db_clone_manager_impl.go:160: Info: Postgres Database clones:
clone/postgres: 2023/11/01 23:04:49.932872 db_clone_manager_impl.go:162: Info: central_active -> &{ 4.3.x-25-gca713734e4 195 186 2023-11-01 16:55:10.18614 +0000 UTC}
clone/postgres: 2023/11/01 23:04:49.932893 db_clone_manager_impl.go:225: Info: GetCloneToMigrate
clone/postgres: 2023/11/01 23:04:49.932933 db_clone_manager_impl.go:169: Info: safeRemove -> central_previous
NOTICE:  database "central_previous" does not exist, skipping
clone/postgres: 2023/11/01 23:04:50.100328 db_clone_manager_impl.go:491: Info: Central has space to create backup for rollback: true, required: 42433593, available: 42468044800 with 0.200000 margin
Migrator: 2023/11/01 23:04:50.100481 log.go:18: Info: Clone to Migrate "", "central_active"
Migrator: 2023/11/01 23:04:50.100512 log.go:13: Info: starting DB compaction
Migrator: 2023/11/01 23:04:50.100590 log.go:13: Info: no existing DB exists. Stopping DB compaction
Migrator: 2023/11/01 23:04:50.100660 log.go:18: Info: Load database = "central_active"
Migrator: 2023/11/01 23:04:50.213475 log.go:13: Info: Successfully connected to central database.
Migrator: 2023/11/01 23:04:50.301602 log.go:18: Info: Migration version from DB = seq_num:195 version:"4.3.x-25-gca713734e4" last_persisted:<seconds:1698857710 nanos:186140000 > min_seq_num:186 .
Migrator: 2023/11/01 23:04:50.301721 log.go:18: Info: version for "central_active" is &{ 4.3.x-25-gca713734e4 195 186 2023-11-01 16:55:10.18614 +0000 UTC}
Migrator: 2023/11/01 23:04:50.301739 log.go:18: Info: In runner.Run
Migrator: 2023/11/01 23:04:50.350540 log.go:18: Info: Migration version from DB = seq_num:195 version:"4.3.x-25-gca713734e4" last_persisted:<seconds:1698857710 nanos:186140000 > min_seq_num:186 .
Migrator: 2023/11/01 23:04:50.350635 log.go:18: Info: Found DB at version 195, which is less than what we expect (196). Running migrations...
migrations/m_195_to_m_196_vuln_request_users: 2023/11/01 23:04:50.764591 migration_impl.go:67: Info: Updated 4 vulnerability exceptions
Migrator: 2023/11/01 23:04:50.768448 log.go:18: Info: Successfully updated DB from version 195 to 196


$ curl -sSk -u "admin:e9s99uQzFToVYSaaWowYetAsM" https://104.154.138.198/v1/cve/requests | jq
{
  "requestInfos": [
    {
      "id": "d0bc53f8-e0bd-4532-a699-45b515a6fb04",
      "name": "AA-231101-1",
      "targetState": "DEFERRED",
      "status": "APPROVED",
      "expired": true,
      "requestor": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [
        {
          "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
          "name": "admin"
        }
      ],
      "createdAt": "2023-11-01T15:42:06.526413935Z",
      "lastUpdated": "2023-11-01T23:04:59.055207796Z",
      "comments": [
        {
          "id": "90db1e85-064f-4b8d-a674-ab5a5d122ae9",
          "message": "defer",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T15:42:06.526431134Z"
        },
        {
          "id": "f09ce9d1-8bd1-410f-b7b2-a9c8873374b0",
          "message": "approved\n",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T23:00:12.600091431Z"
        },
        {
          "id": "f0d87ead-43b8-4210-bd3c-de3b0831841f",
          "message": "[System Generated] Request expired",
          "user": null,
          "createdAt": "2023-11-01T23:04:59.055225938Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "gke.gcr.io",
          "remote": "calico/node",
          "tag": "v3.23.5-gke.10"
        }
      },
      "requesterV2": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approversV2": [
        {
          "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
          "name": "admin"
        }
      ],
      "deferralReq": {
        "expiry": {
          "expiresWhenFixed": true,
          "expiryType": "TIME"
        }
      },
      "cves": {
        "cves": [
          "RHSA-2023:5455"
        ]
      }
    },
    {
      "id": "874550d1-d208-46c5-8780-c02dfa425269",
      "name": "AA-231101-2",
      "targetState": "DEFERRED",
      "status": "PENDING",
      "expired": false,
      "requestor": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [],
      "createdAt": "2023-11-01T15:42:14.572726897Z",
      "lastUpdated": "2023-11-01T15:42:14.572726897Z",
      "comments": [
        {
          "id": "6d8f8b90-bbd9-4af0-ac03-e0b10097e86f",
          "message": "defer",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T15:42:14.572740685Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "gke.gcr.io",
          "remote": "calico/node",
          "tag": ".*"
        }
      },
      "requesterV2": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approversV2": [],
      "deferralReq": {
        "expiry": {
          "expiresWhenFixed": true,
          "expiryType": "TIME"
        }
      },
      "cves": {
        "cves": [
          "CVE-2021-35937"
        ]
      }
    },
    {
      "id": "ca93899f-ca34-4857-b127-90d1aaacece4",
      "name": "AA-231101-3",
      "targetState": "FALSE_POSITIVE",
      "status": "APPROVED",
      "expired": false,
      "requestor": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [
        {
          "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
          "name": "admin"
        }
      ],
      "createdAt": "2023-11-01T22:59:49.263754985Z",
      "lastUpdated": "2023-11-01T23:00:18.940219419Z",
      "comments": [
        {
          "id": "69455fa3-ee75-47a7-b7b7-9a3723892fcf",
          "message": "false positive",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T22:59:49.263776648Z"
        },
        {
          "id": "0cc9585d-9ede-4c2a-ab1e-196f0400690d",
          "message": "approved",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T23:00:18.940228612Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "quay.io",
          "remote": "stackrox-io/central-db",
          "tag": ".*"
        }
      },
      "requesterV2": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approversV2": [
        {
          "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
          "name": "admin"
        }
      ],
      "fpRequest": {},
      "cves": {
        "cves": [
          "CVE-2007-4559"
        ]
      }
    },
    {
      "id": "e9720e3e-11dd-45e7-bd50-e109249a936e",
      "name": "AA-231101-4",
      "targetState": "DEFERRED",
      "status": "PENDING",
      "expired": false,
      "requestor": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [],
      "createdAt": "2023-11-01T23:00:00.854599976Z",
      "lastUpdated": "2023-11-01T23:00:00.854599976Z",
      "comments": [
        {
          "id": "5fa5a0b5-e9b8-4e2d-bd8f-8b484b299e1a",
          "message": "defer",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T23:00:00.854609382Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "quay.io",
          "remote": "stackrox-io/central-db",
          "tag": ".*"
        }
      },
      "requesterV2": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approversV2": [],
      "deferralReq": {
        "expiry": {
          "expiresWhenFixed": true,
          "expiryType": "TIME"
        }
      },
      "cves": {
        "cves": [
          "CVE-2016-4607"
        ]
      }
    }
  ]
}


$ curl -sSk -u "admin:e9s99uQzFToVYSaaWowYetAsM" https://104.154.138.198/v2/vulnerability-exceptions | jq
{
  "exceptions": [
    {
      "id": "d0bc53f8-e0bd-4532-a699-45b515a6fb04",
      "name": "AA-231101-1",
      "targetState": "DEFERRED",
      "status": "APPROVED",
      "expired": true,
      "requester": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [
        {
          "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
          "name": "admin"
        }
      ],
      "createdAt": "2023-11-01T15:42:06.526413935Z",
      "lastUpdated": "2023-11-01T23:04:59.055207796Z",
      "comments": [
        {
          "id": "90db1e85-064f-4b8d-a674-ab5a5d122ae9",
          "message": "defer",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T15:42:06.526431134Z"
        },
        {
          "id": "f09ce9d1-8bd1-410f-b7b2-a9c8873374b0",
          "message": "approved\n",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T23:00:12.600091431Z"
        },
        {
          "id": "f0d87ead-43b8-4210-bd3c-de3b0831841f",
          "message": "[System Generated] Request expired",
          "user": null,
          "createdAt": "2023-11-01T23:04:59.055225938Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "gke.gcr.io",
          "remote": "calico/node",
          "tag": "v3.23.5-gke.10"
        }
      },
      "deferralRequest": {
        "expiry": {
          "expiryType": "TIME",
          "expiresOn": null
        }
      },
      "cves": [
        "RHSA-2023:5455"
      ]
    },
    {
      "id": "874550d1-d208-46c5-8780-c02dfa425269",
      "name": "AA-231101-2",
      "targetState": "DEFERRED",
      "status": "PENDING",
      "expired": false,
      "requester": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [],
      "createdAt": "2023-11-01T15:42:14.572726897Z",
      "lastUpdated": "2023-11-01T15:42:14.572726897Z",
      "comments": [
        {
          "id": "6d8f8b90-bbd9-4af0-ac03-e0b10097e86f",
          "message": "defer",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T15:42:14.572740685Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "gke.gcr.io",
          "remote": "calico/node",
          "tag": ".*"
        }
      },
      "deferralRequest": {
        "expiry": {
          "expiryType": "TIME",
          "expiresOn": null
        }
      },
      "cves": [
        "CVE-2021-35937"
      ]
    },
    {
      "id": "ca93899f-ca34-4857-b127-90d1aaacece4",
      "name": "AA-231101-3",
      "targetState": "FALSE_POSITIVE",
      "status": "APPROVED",
      "expired": false,
      "requester": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [
        {
          "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
          "name": "admin"
        }
      ],
      "createdAt": "2023-11-01T22:59:49.263754985Z",
      "lastUpdated": "2023-11-01T23:00:18.940219419Z",
      "comments": [
        {
          "id": "69455fa3-ee75-47a7-b7b7-9a3723892fcf",
          "message": "false positive",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T22:59:49.263776648Z"
        },
        {
          "id": "0cc9585d-9ede-4c2a-ab1e-196f0400690d",
          "message": "approved",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T23:00:18.940228612Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "quay.io",
          "remote": "stackrox-io/central-db",
          "tag": ".*"
        }
      },
      "falsePositiveRequest": {},
      "cves": [
        "CVE-2007-4559"
      ]
    },
    {
      "id": "e9720e3e-11dd-45e7-bd50-e109249a936e",
      "name": "AA-231101-4",
      "targetState": "DEFERRED",
      "status": "PENDING",
      "expired": false,
      "requester": {
        "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
        "name": "admin"
      },
      "approvers": [],
      "createdAt": "2023-11-01T23:00:00.854599976Z",
      "lastUpdated": "2023-11-01T23:00:00.854599976Z",
      "comments": [
        {
          "id": "5fa5a0b5-e9b8-4e2d-bd8f-8b484b299e1a",
          "message": "defer",
          "user": {
            "id": "sso:4df1b98c-24ed-4073-a9ad-356aec6bb62d:admin",
            "name": "admin"
          },
          "createdAt": "2023-11-01T23:00:00.854609382Z"
        }
      ],
      "scope": {
        "imageScope": {
          "registry": "quay.io",
          "remote": "stackrox-io/central-db",
          "tag": ".*"
        }
      },
      "deferralRequest": {
        "expiry": {
          "expiryType": "TIME",
          "expiresOn": null
        }
      },
      "cves": [
        "CVE-2016-4607"
      ]
    }
  ]
}
```

### CODE REVIEW

File: central/convert/v2tostorage/vuln_exception_service.go [hunk 1]
```diff
@@ -236,3 +242,30 @@ func requestExpiryType(t v2.ExceptionExpiry_ExpiryType) storage.RequestExpiry_Ex
 		return storage.RequestExpiry_TIME
 	}
 }
+
+func requester(user *storage.SlimUser) *storage.Requester {
```
Comment (theencee) [inline central/convert/v2tostorage/vuln_exception_service.go:246]: Can you move this to a shared util since this code is duplicated twice?

**Review by theencee — APPROVED** [review]

**Comment by dashrews78** [comment]
> This is because the requester and approvers fields shared the same type storage.SlimUser and we do not support different search labels on the same message or message hierarchy.

Seems like something we should give some thought to at some point.  Do we/Should we have a ticket to extend support for or investigate supporting this scenario in the future?

File: central/convert/storagetov2/vuln_exception_service.go [hunk 2]
```diff
@@ -31,8 +31,8 @@ func VulnerabilityException(vulnRequest *storage.VulnerabilityRequest) *v2.Vulne
 		TargetState: convertVulnerabilityState(vulnRequest.GetTargetState()),
 		Status:      exceptionStatus(vulnRequest.GetStatus()),
 		Expired:     vulnRequest.GetExpired(),
-		Requester:   convertUser(vulnRequest.GetRequestor()),
-		Approvers:   convertUsers(vulnRequest.GetApprovers()),
+		Requester:   requester(vulnRequest.GetRequesterV2()),
```
Comment (dashrews78) [inline central/convert/storagetov2/vuln_exception_service.go:34]: This caused me to go down a rabbit hole of "requester" vs "requestor".  I learned 2 things:
1. I concur with the change
2. English is painful

**Review by dashrews78 — APPROVED** [review]

File: central/convert/v2tostorage/vuln_exception_service.go (hunk 1 above)
Comment (theencee) [reply central/convert/v2tostorage/vuln_exception_service.go:246]: central/convert/storagetov2/vuln_exception_service.go

**Comment by md2119** [comment]
> > This is because the requester and approvers fields shared the same type storage.SlimUser and we do not support different search labels on the same message or message hierarchy.
> 
> Seems like something we should give some thought to at some point. Do we/Should we have a ticket to extend support for or investigate supporting this scenario in the future?

I have a couple of learnings from my recent sprint of writing migrations, which I plan on sharing with the broader team since it could be useful for everyone.

**Pull request merged**